### PR TITLE
doc: final report added

### PR DESCRIPTION
**Pull request by DevanshD3**
This is the final report of the project recommendation system for GSoC'21.

### CODE REVIEW

File: content/posts/Devash_Recommendation_System_Final_Report.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+title: 'Project Recommendation System: Final Report (GSoC 2021- Devansh Dixit)'
+date: 2021-08-13T21:32:37+05:30
+draft: false
+author: Devansh Dixit
+type: post
+---
+
+# Recommendation System
+
+The recommendation system is the first ML project for [CircuitVerse](https://github.com/CircuitVerse/CircuitVerse). The basic idea is to recommend similar circuits (which are also popular) to the users to increase user engagement and reach of good projects. In this report, I have explained every decision or technology in simple words to give a better understanding and to give an idea of why we did, what we did.
```
Comment (biswesh456) [inline content/posts/Devash_Recommendation_System_Final_Report.md:11]: Just say this project aims to add the functionality of a  recommendation system in first line.

File: content/posts/Devash_Recommendation_System_Final_Report.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+### All my code files 💻🎉 [here](https://github.com/DevanshD3/CircuitVerse/tree/recommendation-system/recommendation%20system).
+
+## Why use an unsupervised learning-based model?
+
+The data that was available for us (check out the [database schema](https://github.com/CircuitVerse/CircuitVerse/blob/master/db/schema.rb#L275)) to use didn't have a response variable (typically a X and a Y to train and test the model and improve the accuracy). Therefore, we had to rely on the usage of unsupervised learning-based algorithms to find the intrinsic pattern and hidden structures in the data and present it to the user.
```
Comment (biswesh456) [inline content/posts/Devash_Recommendation_System_Final_Report.md:17]: replace response with target

File: content/posts/Devash_Recommendation_System_Final_Report.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+- Each document is modeled as a multinomial distribution of topics and each topic is modeled as a multinomial distribution of words.
+- LDA assumes that every chunk of text we feed into it will contain words that are somehow related. Therefore choosing the right corpus of data is crucial.
+- It also assumes documents are produced from a mixture of topics. Those topics then generate words based on their probability distribution.
+
+Simply put, LDA creates a distribution of words and packs them together as a topic (the number of topics is chosen by the user). When a new sentence is fed into the model, based on the words present in the new sentence, we get a distribution of the sentence over all the topics. The new dimension of each vector becomes the total number of topics hence drastically reducing the dimensions with better filtering.
+In our recommendation system, the log-likelihood was the least for 10 topics for the given dataset.
```
Comment (biswesh456) [inline content/posts/Devash_Recommendation_System_Final_Report.md:48]: Add bit more stating how each topic is considered as a dimension and the corresponding socres for each topic is used as the new vector representation thus getting a compact form of representation. This is how we do dimensionality reduction which later helps in finding better neighbours in kd trees

File: content/posts/Devash_Recommendation_System_Final_Report.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+![Topics 5 to 50](/images/DevanshD3-GSoC'21/MT.png)
+**Topics 5 to 50**
+![Zoomed for topics 5 to 20](/images/DevanshD3-GSoC'21/LT.png)
+**Zoomed for topics 5 to 20**
+
+## K-D Trees
+
+After getting the probability distribution of each project we can compare the items and thus enhancing the 2nd layer of the model. Item distance being the best collaborative filtering approach gives us the most similar items. However, this will be a computationally expensive method with time complexity of `O(N^2 log(N))`, and training the model will take days. We also won't be able to use this method for new project recommendations in the online mode because of the high runtime.
```
Comment (biswesh456) [inline content/posts/Devash_Recommendation_System_Final_Report.md:57]: You havent mentioned the two layered model in the beginning

**Review by ShreyaPrasad1209 — CHANGES_REQUESTED** [review]
This looks good! Few changes I would recommend :

1. Font size of title is smaller than sub-headings. It should be vice-versa.
2. "All my codes here" can be included in first paragraph. 
3. Increase space between title and first sub-heading. Same goes for every heading.
4.  Increase font size of content under "My other content pieces" heading

**Comment by DevanshD3** [comment]
@ShreyaPrasad1209 I have incorporated points 1,2,3 successfully but point number 4, the default size for an unordered list is fixed and cannot be changed explicitly. Request you to merge the blog with the main repo.

File: content/posts/Devash_Recommendation_System_Final_Report.md (hunk 4 above)
Comment (biswesh456) [reply content/posts/Devash_Recommendation_System_Final_Report.md:77]: Did you mention it?

Comment (DevanshD3) [reply content/posts/Devash_Recommendation_System_Final_Report.md:77]: Yes yes I  have incorporated all the changes

Comment (DevanshD3) [reply content/posts/Devash_Recommendation_System_Final_Report.md:77]: https://deploy-preview-92--cvblog.netlify.app/posts/devash_recommendation_system_final_report/ 
this is the final report

**Review by ShreyaPrasad1209 — APPROVED** [review]

**Pull request merged**